erdpy/projects/report/report_creator.py

```diff
@@ -0,0 +1,81 @@
+
+import itertools
+import operator
+from pathlib import Path
+from typing import Iterable, List, Tuple
+from erdpy import guards
+from erdpy.projects.report.data.folder_report import FolderReport
+from erdpy.projects.report.data.option_results import OptionResults
+from erdpy.projects.report.data.report import Report
+from erdpy.projects.report.data.wasm_report import WasmReport
+from erdpy.projects.report.data.project_report import ProjectReport
+from erdpy.projects.core import load_project
+from erdpy.projects.project_base import remove_suffix
+from erdpy.projects.project_rust import ProjectRust
+
+from erdpy.projects.report.options.report_option import ReportOption
+from erdpy.projects.report.options.twiggy_paths_check import run_twiggy_paths
+
+
+class ReportCreator:
+    def __init__(self, options: List[ReportOption], skip_build: bool, skip_twiggy: bool) -> None:
+        self.options = options
+        self.skip_build = skip_build
+        self.skip_twiggy = skip_twiggy
+        self.require_twiggy_paths = any(option.requires_twiggy_paths() for option in self.options)
+
+
+    def apply_option(self, option: ReportOption, wasm_path: Path) -> OptionResults:
+        result = option.apply(wasm_path)
+        return OptionResults(option.name, [result])
+
+
+    def create_wasm_report(self, wasm_path: Path, twiggy_requirements_met: bool) -> WasmReport:
```

Comment:
Perhaps reorder methods, from `large item` to `small item` - that is, **folder** is largest, then **project** etc.

Perhaps make all of them private, with the exception of `create_report` (the only one called from outside, if not mistaken)?